pagopa/io-app
# chore: [IOBP-615] Add new `paymentCard` component to `PaymentsMethodDetailsScreen`

## Short description
This PR refactors the PaymentsMethodDetailsScreen component with an updated UI. This PR also adds the wallet card removal if user deletes the method.

## List of changes proposed in this pull request
- Fixed an issue with the expiry date format in `PaymentCard` component
- Added `PaymentCard` component inside `PaymentsMethodDetailsScreen`
- Simplified rendering login inside `PaymentsMethodDetailsScreen`

## How to test
With the `io-dev-api-server`, navigate to the details screen of a payment method. Check that the components render like in the preview video below

## Preview

<video src="https://github.com/pagopa/io-app/assets/6160324/a8f3bfed-3819-44b1-beb8-4dff8f4a768f" />
